src/domain/core/src/services/config.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use internal_error::{InternalError, ResultIntoInternal};
+use url::Url;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct Config {
+    pub protocols: Protocols,
```

Comment:
NB. `config-rs` supports hierarchical serialization

Reply:
There are several libs that serve this purpose,

Reply:
yes, I took this one as a base from your comment https://github.com/kamu-data/kamu-node/issues/45#issuecomment-1981886559